src/libs/intents/localMethods.ts

```diff
@@ -121,7 +121,7 @@ export const localMethods = (
     hideSplashScreen: nativeMethodWrapper(hideSplashScreen),
     logout: () => asyncLogout(client),
     openApp: (href, app, iconParams) =>
-      openApp(RootNavigation, href, app, iconParams),
+      openApp(client, RootNavigation, href, app, iconParams),
```

Comment:
even if we're in TS (thankfully here), I think that's starting to be a lot of arguments for that function.

This might reveal a bigger issue with the openApp systems, why do we need so many parameters?

Reply:
This method is called from the outside of the React Context. So we need to get access to the router and to the client. We can inject them by magic by using some kind of middleware, but I always prefer readability / verbosity versus magic. 

Or... we change to be event based. Like that we can listen to a specific message in a React context and we'll not have to pass those 2 args. But this is not something I want to go in.